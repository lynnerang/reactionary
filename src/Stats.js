import React from 'react';
import './Stats.scss';

function Stats(props) {
  return (
    <>
      <h2>Stats</h2>
      <div className='stat'>
      <i className='fas fa-check-double'></i>
        <div>
          <h3 className='stats-heading'>Total Games Completed</h3>
          <p>You have completed a total of {props.gameCount} games!</p>
        </div>
      </div>
      <div className='stat'>
        <div><i className='fas fa-trophy'></i></div>
        <div></div>
        <div>
          <h3 className='stats-heading'>High Score</h3>
          <p>Your best game was answering {props.highScore.totalCards} questions with {props.highScore.guesses} guesses!</p>
        </div>
      </div>
    </>
  )
}

export default Stats;