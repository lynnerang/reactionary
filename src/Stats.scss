@import './variables.scss';

.stats-heading {
  margin-bottom: 15px;
}

.stat {
  padding: 30px;
  display: flex;
  align-items: center;

  .fas {
    font-size: 60px;
    margin-right: 20px;
  }

  .fa-check-double {
    color: $primary-color;
  }

  .fa-trophy {
    color: rgba(255, 196, 0, 0.653)
  }
}