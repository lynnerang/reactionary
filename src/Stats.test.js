import React from 'react';
import Stats from './Stats.js';
import { shallow } from 'enzyme';

const props = {
  highScore: {guesses: 6, totalCards: 4},
  gameCount: 2
}


describe('Stats', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <Stats {...props}
      />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});