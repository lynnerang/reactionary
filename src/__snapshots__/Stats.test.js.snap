// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Should match the snapshot 1`] = `
<Fragment>
  <h2>
    Stats
  </h2>
  <div
    className="stat"
  >
    <i
      className="fas fa-check-double"
    />
    <div>
      <h3
        className="stats-heading"
      >
        Total Games Completed
      </h3>
      <p>
        You have completed a total of 
        2
         games!
      </p>
    </div>
  </div>
  <div
    className="stat"
  >
    <div>
      <i
        className="fas fa-trophy"
      />
    </div>
    <div />
    <div>
      <h3
        className="stats-heading"
      >
        High Score
      </h3>
      <p>
        Your best game was answering 
        4
         questions with 
        6
         guesses!
      </p>
    </div>
  </div>
</Fragment>
`;
